addons/mail/static/src/core/common/suggestion_hook.js

```diff
@@ -188,16 +198,30 @@ export class UseSuggestion {
             this.composer.cannedResponses.push(option.cannedResponse);
         }
         if (this.comp.composerService.htmlEnabled) {
-            this.comp.editor.shared.dom.insert(option.label);
+            let inlineElement;
+            if (option.partner) {
+                inlineElement = generateMentionElement("partner", option.partner);
+            } else if (option.isSpecial) {
+                inlineElement = generateMentionElement("special", option);
+            } else if (option.role) {
+                inlineElement = generateMentionElement("role", option.role);
+            } else if (option.thread) {
+                inlineElement = generateMentionElement("thread", option.thread);
+            } else {
+                inlineElement = createTextNode(option.label);
+            }
+            this.comp.editor.shared.dom.insert(inlineElement);
+            const [anchorNode, anchorOffset] = rightPos(inlineElement);
+            this.comp.editor.shared.selection.setSelection({ anchorNode, anchorOffset });
             this.comp.editor.shared.history.addStep();
         } else {
             // remove the user-typed search delimiter
             this.composer.composerText =
                 this.composer.composerText.substring(0, position) +
                 this.composer.composerText.substring(this.composer.selection.end);
-            this.clearSearch();
```

Comment:
This is a bugfix right?
Is there already something broken before this PR?

Reply:
nope, there is only canned response before this PR and it is not triggering bug because the search will be invalid after inserting. But in this PR, this is triggered but @marc demo will make the navi list stay open.

But technically, this is a fix.